sdk/storage/Azure.Storage.DataMovement.Blobs/README.md

```diff
@@ -40,7 +40,178 @@ az storage account create --name MyStorageAccount --resource-group MyResourceGro
+    DataTransfer dataTransfer = await transferManager.StartTransferAsync(
+        sourceResource: new LocalDirectoryStorageResourceContainer(sourcePath),
+        destinationResource: new BlobDirectoryStorageResourceContainer(container, Randomize("sample-blob-directory")),
+        transferOptions: options);
+}
+```
+
+```C# Snippet:UploadDirectory_CompletedEventHandler
+    // Create BlobTransferManager with event handler in Options bag
+    TransferManagerOptions options = new TransferManagerOptions();
+    ContainerTransferOptions containerTransferOptions = new ContainerTransferOptions();
+    containerTransferOptions.SingleTransferCompleted += (SingleTransferCompletedEventArgs args) =>
+    {
+        // Customers like logging their own exceptions in their production environments.
+        using (StreamWriter logStream = File.AppendText(logFile))
+        {
+            logStream.WriteLine($"File Completed Transfer: {args.SourceResource.Path}");
+        }
+        return Task.CompletedTask;
+    };
+    TransferManager transferManager = new TransferManager(options);
+
+    // Create simple transfer directory upload job which uploads the directory and the contents of that directory
+    DataTransfer uploadDirectoryJobId = await transferManager.StartTransferAsync(
+        new LocalDirectoryStorageResourceContainer(sourcePath),
+        new BlobDirectoryStorageResourceContainer(container, Randomize("sample-blob-directory")));
+}
+```
+
+```C# Snippet:UploadDirectory_EventHandler_SasAsync
+    // Create BlobTransferManager with event handler in Options bag
+    TransferManagerOptions options = new TransferManagerOptions();
+    ContainerTransferOptions containerTransferOptions = new ContainerTransferOptions();
+    containerTransferOptions.TransferStatus += (TransferStatusEventArgs args) =>
+    {
+        if (args.StorageTransferStatus == StorageTransferStatus.Completed)
+        {
+            // Customers like logging their own exceptions in their production environments.
+            using (StreamWriter logStream = File.AppendText(logFile))
+            {
+                logStream.WriteLine($"Our transfer has completed!");
+            }
+        }
+        return Task.CompletedTask;
+    };
+    containerTransferOptions.TransferFailed += (TransferFailedEventArgs args) =>
+    {
+        // Customers like logging their own exceptions in their production environments.
+        using (StreamWriter logStream = File.AppendText(logFile))
+        {
+            // Specifying specific resources that failed, since its a directory transfer
+            // maybe only one file failed out of many
+            logStream.WriteLine($"Exception occured with TransferId: {args.TransferId}," +
+                $"Source Resource: {args.SourceResource.Path}, +" +
+                $"Destination Resource: {args.DestinationResource.Path}," +
+                $"Exception Message: {args.Exception.Message}");
+        }
+        return Task.CompletedTask;
+    };
```

Comment:
same here. make this it's own snippet on how to add delegates to the failed event.